rsciio/jobin_yvon/api.py

```diff
@@ -0,0 +1,819 @@
+        try:
+            self.original_metadata["experimental_setup"][
+                "measurement_type"
+            ] = self._measurement_type
+        except AttributeError:
+            pass  # pragma: no cover
+        try:
+            self.original_metadata["experimental_setup"]["title"] = self._title
+        except AttributeError:
+            pass  # pragma: no cover
+        try:
+            self.original_metadata["experimental_setup"][
+                "rotation angle (rad)"
+            ] = self._angle
+        except AttributeError:
+            pass
+        self._clean_up_metadata()
+
+    def _set_signal_type(self, xml_element):
+        """Sets signal type and units based on metadata from file.
+
+        Extra method, because this information is stored seperate from the rest of the metadata.
+
+        Parameters
+        ----------
+        xml_element: xml.etree.ElementTree.Element
+            Head level metadata element.
+        """
+        for child in xml_element:
+            id = self._get_id(child)
+            ## contains also intensity-minima/maxima-values for each data-row (ignored by this reader)
+            if id == "0x6D707974":
+                self.original_metadata["experimental_setup"]["signal type"] = child.text
+            if id == "0x7C696E75":
+                self.original_metadata["experimental_setup"][
+                    "signal units"
+                ] = child.text
+
+    def _get_scale(self, array, name):
+        """Get scale for navigation/signal axes.
+
+        Furthermore, checks whether the axis is uniform. Throws warning when this not the case.
+        This check is performed by comparing the difference between the first 2 and last 2 values.
+        The decision to use a non-uniform/uniform data-axis is left to the user
+        (use_uniform_wavelength_axis).
+
+        Parameters
+        ----------
+        array: np.ndarray
+            Contains the axes data points.
+        name: Str
+            Name of the axis.
+        """
+        scale = np.abs(array[0] - array[-1]) / (array.size - 1)
+        if array.size >= 3:
+            abs_diff_begin = np.abs(array[0] - array[1])
+            abs_diff_end = np.abs(array[-1] - array[-2])
+            rel_diff_compare = np.abs(abs_diff_begin - abs_diff_end) / scale
+            if rel_diff_compare > 0.01 and self._use_uniform_signal_axis:
+                _logger.warning(
+                    f"The relative variation of the {name}-axis-scale ({rel_diff_compare:.3f}%) exceeds 1%.\n"
+                    "                              "
+                    f"Difference between first 2 entrys: {abs_diff_begin:.3f}.\n"
+                    "                              "
+                    f"Difference between last 2 entrys: {abs_diff_end:.3f}.\n"
+                    "                              "
+                    "Using a non-uniform-axis is recommended."
+                )
```

Comment:
Here, is there any specific reason to compare the first / last two entries? 

I would intuitively warn if the max error exceeds the threshold.
```python
rel_diff_compare  = (np.diff(array) - scale)/scale
if rel_diff_compare > 0.01 and self._use_uniform_signal_axis:
    ...
```

Reply:
In the test datasets (array[i+1] - array[i]) decreases from the first 2 to the last 2 values. But checking for the max error makes generally more sense. So I will change that.

Reply:
> Also, a 1% threshold certainly seems very large to me but feel free to let it that way if you think that's the way to go

In the beginning I thought that as well, but even with that threshold it would warn on most of our test datasets. That is why the idea is to default to a non-uniform axis, but allow the loading as a uniform axis as an option.

Reply:
Ok :)

Defaulting to a non-uniform axis is an even better option I think. 

I am not sure how general is the scope of the Jobin Yvon XML file format. If it's dedicated to spectroscopy then cubic axes (1/cubic for the energy-like representation) would cover everything.